libraries/AP_Proximity/AP_Proximity_ARS408_CAN.cpp

```diff
@@ -0,0 +1,388 @@
+#include <AP_HAL/AP_HAL.h>
+#include "AP_Proximity_ARS408_CAN.h"
+#include <ctype.h>
+#include <algorithm>
+#include <string>
+#include <limits>
+#include <memory>
+#include <utility>
+
+#include <AP_AHRS/AP_AHRS.h>
+#include <AP_GPS/AP_GPS.h>
+
+#if HAL_MAX_CAN_PROTOCOL_DRIVERS
+
+#define RP_DEBUG_LEVEL 0
+
+#if RP_DEBUG_LEVEL
+  #include <GCS_MAVLink/GCS.h>
+  #define Debug(level, fmt, args ...)  do { if (level <= RP_DEBUG_LEVEL) { gcs().send_text(MAV_SEVERITY_INFO, fmt, ## args); } } while (0)
+#else
+  #define Debug(level, fmt, args ...)
+#endif
+
+ AP_Proximity_ARS408_CAN::AP_Proximity_ARS408_CAN(AP_Proximity &_frontend, AP_Proximity::Proximity_State &_state):
+    CANSensor("RS408"),
+    AP_Proximity_Backend(_frontend,_state)
+    {
+        register_driver(AP_CANManager::Driver_Type_ARS408);
+        radar_cfg_msg.data.RadarCfg_MaxDistance_valid = 0;
+        radar_cfg_msg.data.RadarCfg_SensorID_valid = 0;
+        radar_cfg_msg.data.RadarCfg_RadarPower_valid = 0;
+        radar_cfg_msg.data.RadarCfg_OutputType_valid = 0;
+        radar_cfg_msg.data.RadarCfg_SendQuality_valid = 0;
+        radar_cfg_msg.data.RadarCfg_SendExtInfo_valid = 0;
+        radar_cfg_msg.data.RadarCfg_SortIndex_valid = 0;
+        radar_cfg_msg.data.RadarCfg_CtrlRelay_valid = 0;
+        radar_cfg_msg.data.RadarCfg_StoreInNVM_valid = 0;
+        radar_cfg_msg.data.RadarCfg_RCS_Threshold_valid = 0;
```

Comment:
This is wasted flash space, unless you're managing to put this on the stack somehow :-)

https://ardupilot.org/dev/docs/style-guide.html#implicit-zeroing-of-memory